spec/models/user_spec.rb

```diff
@@ -160,4 +162,98 @@
       end
     end
   end
+
+  describe '#visible_assessments' do
+
+    context "when there are no assessments" do
+      let(:new_user) {create :user, user_name: 'testusername', first_name: 'testfname',
+                               last_name: "testlname", type: "Student"}
+      it "should return an empty list" do
+        expect(new_user.visible_assessments()).to match_array([])
+      end
+    end
+    context "when there are assessments" do
+      before :each do
+        @new_section = create(:section)
+        @assignment_visible = create(:assignment,
+                                    due_date: 2.days.from_now,
+                                    assignment_properties_attributes: { section_due_dates_type: true })
+        @section_due_date_visible = create(:section_due_date, assessment: @assignment_visible,
+                                           section: @new_section,
+                                           due_date: 2.days.from_now,
+                                           is_hidden: false)
+        @assignment_hidden = create(:assignment,
+                                     due_date: 2.days.from_now,
+                                     assignment_properties_attributes: { section_due_dates_type: true })
+        @section_due_date_hidden = create(:section_due_date, assessment: @assignment_hidden,
+                                           section: @new_section,
+                                           due_date: 2.days.from_now,
+                                           is_hidden: true)
+      end
+      context "when section_due_dates disabled" do
+        let(:new_user_2) {create :user, user_name: 'testusername', first_name: 'testfname',
+                                 last_name: "testlname", type: "Student"}
+        it 'does return all unhiddden assignments' do
+          expect(new_user_2.visible_assessments()).to match_array([@assignment_visible, @assignment_hidden])
```

Comment:
The `match_array` matcher is good if you care about order. In this case I would look at the `contains_exactly` matcher instead (https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/contain-exactly-matcher)

That way your tests won't accidentally fail if the query returns things in a different order.